### PR TITLE
Improve diagnostics when PROCESSOR_ARCHITEW6432 is set to bogus

### PR DESCRIPTION
**Pull request by BillyONeal**
.... such as "%processor_architew6432%", as reported by Cameron DaCamara.

### CODE REVIEW

File: include/vcpkg/base/messages.h [hunk 1]
```diff
@@ -12,8 +12,6 @@
 
 namespace vcpkg::msg
 {
-    struct LocalizedString;
```
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:15]: This is already forward declared in <vcpkg/base/fwd/messages.h>

File: src/vcpkg/base/system.cpp [hunk 2]
```diff
@@ -50,11 +70,34 @@ namespace vcpkg
...
+
+            msg::print(Color::warning, msgProcessorArchitectureW6432Malformed, msg::value = *w6432);
+        }
+
+        const auto raw_processor_architecture = get_environment_variable("PROCESSOR_ARCHITECTURE");
+        const auto processor_architecture = raw_processor_architecture.get();
+        if (!processor_architecture)
+        {
+            msg::print(Color::error, msgProcessorArchitectureMissing);
+            Checks::exit_with_code(VCPKG_LINE_INFO, 1);
```
Comment (BillyONeal) [inline src/vcpkg/base/system.cpp:90]: I asked @strega-nil if this is the intended usage for now.

Comment (strega-nil) [reply src/vcpkg/base/system.cpp:90]:
```suggestion
            Checks::exit_with_message(VCPKG_LINE_INFO, msgProcessorArchitectureMissing);
```

**Comment by BillyONeal** [comment]
I asked @strega-nil for help in figuring out why the generated message map is unhappy.

**Pull request merged**